components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/NatsInjectHandler.java

```
@@ -0,0 +1,151 @@
+            MessageContext axis2MsgCtx = ((org.apache.synapse.core.axis2.Axis2MessageContext) msgCtx)
+                    .getAxis2MessageContext();
+
+            // Check if the reply to field is null and send the response the the NATS publisher from the Respond Mediator
+            if (replyTo != null) {
+                msgCtx.setProperty(InboundEndpointConstants.INBOUND_ENDPOINT_RESPONSE_WORKER, new NatsReplySender(replyTo, connection));
+            }
+
+            // Determine the message builder to use
+            Builder builder;
+            if (contentType == null) {
+                printDebugLog("No content type specified. Using SOAP builder.");
+                builder = new SOAPBuilder();
+            } else {
+                int index = contentType.indexOf(';');
+                String type = index > 0 ? contentType.substring(0, index) : contentType;
+                builder = BuilderUtil.getBuilderFromSelector(type, axis2MsgCtx);
+                if (builder == null) {
+                    log.warn("No message builder found for type '" + type + "'. Falling back to SOAP.");
+                    builder = new SOAPBuilder();
+                }
+            }
+            // set the message payload to the message context
+            InputStream in = new AutoCloseInputStream(new ByteArrayInputStream((byte[]) object));
+            OMElement documentElement = builder.processDocument(in, contentType, axis2MsgCtx);
+            msgCtx.setEnvelope(TransportUtils.createSOAPEnvelope(documentElement));
+            // Inject the message to the sequence.
+            if (StringUtils.isEmpty(injectingSeq)) {
+                log.error("Sequence name not specified. Sequence : " + injectingSeq);
+                return false;
+            }
+            SequenceMediator seq = (SequenceMediator) synapseEnvironment.getSynapseConfiguration().getSequence(injectingSeq);
+            if (seq != null) {
```

same as above, we need to throw an exception in either case